python/pip_install/pip_repository.bzl

```diff
@@ -669,6 +700,15 @@ See the example in rules_python/examples/pip_parse_vendored.
 )
 
 def _whl_library_impl(rctx):
+    target_platforms = []
+    if len(rctx.attr.target_platforms) == 1 and rctx.attr.target_platforms[0] == "all":
+        target_platforms = _ALL_PLATFORMS
+    elif rctx.attr.target_platforms:
+        target_platforms = rctx.attr.target_platforms
+        for p in target_platforms:
+            if p not in _ALL_PLATFORMS:
```

Comment:
What defines the set of known/valid platforms?

I guess it's somewhat limited by what we can generate constraints for. But, ignoring that aspect.

It looks like it's technically unbounded? After going through a few peps, I found this: https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#platform-compatibility-tags

We probably want to link to that somewhere in the comments? Probably over in the wheel.py, since that's where the "parsing" of the whl platform tag is converted to the strings starlark understands?

Reply:
Yeah, technically it is unbouded and in theory, we could just do all of the combinations of `OS` and `Arch` enums.

Reply:
I have removed the starlark validation and left only the validation in Python.